diff --git a/config/BUILDCONFIG.gn b/config/BUILDCONFIG.gn
index 387fdb279..e9c87849f 100644
--- a/config/BUILDCONFIG.gn
+++ b/config/BUILDCONFIG.gn
@@ -570,12 +570,12 @@ if (is_win) {
       [ "//build/config/win:default_cygprofile_instrumentation" ]
 }
 
-if (is_clang && !is_nacl) {
-  default_compiler_configs += [
-    "//build/config/clang:find_bad_constructs",
-    "//build/config/clang:extra_warnings",
-  ]
-}
+#if (is_clang && !is_nacl) {
+#  default_compiler_configs += [
+#    "//build/config/clang:find_bad_constructs",
+#    "//build/config/clang:extra_warnings",
+#  ]
+#}
 
 # Debug/release-related defines.
 if (is_debug) {
diff --git a/config/android/BUILD.gn b/config/android/BUILD.gn
index 5ebde213f..e9e4fb847 100644
--- a/config/android/BUILD.gn
+++ b/config/android/BUILD.gn
@@ -148,11 +148,14 @@ config("runtime_library") {
     if (is_component_build || is_asan) {
       libs += [ "c++_shared" ]
     } else {
-      libs += [ "c++_static" ]
+      libs += [ "c++_shared" ]
     }
     libs += [ "c++abi" ]
   }
-  libs += [ "android_support" ]
+
+  if (current_cpu == "arm" || current_cpu == "x86") {
+    libs += [ "android_support" ]
+  }
 
   # arm builds of libc++ starting in NDK r12 depend on unwind.
   if (current_cpu == "arm") {
@@ -179,6 +182,8 @@ config("runtime_library") {
   if (current_cpu != "mipsel" && current_cpu != "mips64el") {
     ldflags += [ "-Wl,--warn-shared-textrel" ]
   }
+
+  ldflags += [ "-Wl,--version-script=" + rebase_path("//jsi/version-script/makev8jsi.lst") ] 
 }
 
 config("executable_config") {
diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index b34d8fda1..e9751de1a 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1418,6 +1418,8 @@ config("default_warnings") {
       cflags += [ "-Wno-psabi" ]
     }
 
+    cflags += [ "-Wno-deprecated-declarations" ]
+
     if (!is_clang) {
       cflags_cc += [
         # See comment for -Wno-c++11-narrowing.
diff --git a/config/compiler/compiler.gni b/config/compiler/compiler.gni
index 4a6fd985f..7b6e862f8 100644
--- a/config/compiler/compiler.gni
+++ b/config/compiler/compiler.gni
@@ -79,6 +79,8 @@ exclude_unwind_tables =
     (is_chrome_branded && is_official_build) ||
     (is_chromecast && !is_cast_desktop_build && !is_debug && !is_fuchsia)
 
+exclude_unwind_tables = true
+
 # If true, optimize for size. Does not affect windows builds.
 # Linux & Mac favor speed over size.
 # TODO(brettw) it's weird that Mac and desktop Linux are different. We should
