diff --git a/src/objects.cc b/src/objects.cc
index d4af74b2bd..9897813652 100644
--- a/src/objects.cc
+++ b/src/objects.cc
@@ -8664,24 +8664,24 @@ Maybe<bool> JSObject::PreventExtensionsWithTransition(
         PrototypeIterator::GetCurrent<JSObject>(iter), should_throw);
   }
 
-  if (object->map()->has_named_interceptor() ||
-      object->map()->has_indexed_interceptor()) {
-    MessageTemplate::Template message = MessageTemplate::kNone;
-    switch (attrs) {
-      case NONE:
-        message = MessageTemplate::kCannotPreventExt;
-        break;
-
-      case SEALED:
-        message = MessageTemplate::kCannotSeal;
-        break;
-
-      case FROZEN:
-        message = MessageTemplate::kCannotFreeze;
-        break;
-    }
-    RETURN_FAILURE(isolate, should_throw, NewTypeError(message));
-  }
+  // if (object->map()->has_named_interceptor() ||
+  //     object->map()->has_indexed_interceptor()) {
+  //   MessageTemplate::Template message = MessageTemplate::kNone;
+  //   switch (attrs) {
+  //     case NONE:
+  //       message = MessageTemplate::kCannotPreventExt;
+  //       break;
+
+  //     case SEALED:
+  //       message = MessageTemplate::kCannotSeal;
+  //       break;
+
+  //     case FROZEN:
+  //       message = MessageTemplate::kCannotFreeze;
+  //       break;
+  //   }
+  //   RETURN_FAILURE(isolate, should_throw, NewTypeError(message));
+  // }
 
   Handle<NumberDictionary> new_element_dictionary;
   if (!object->HasFixedTypedArrayElements() &&
