diff --git a/BUILD.gn b/BUILD.gn
index c6a58776cd..bfac132f3f 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -3414,6 +3414,12 @@ if (is_component_build) {
   }
 }
 
+group("jsi") {
+  deps = [
+    "jsi:v8jsi",
+  ]
+}
+
 v8_executable("d8") {
   sources = [
     "$target_gen_dir/d8-js.cc",
diff --git a/gni/v8.gni b/gni/v8.gni
index a5312f76ac..c093d0f48b 100644
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -124,9 +124,9 @@ if (target_cpu == "mipsel" || target_cpu == "mips64el" ||
   v8_add_configs += [ "//build/config/gcc:rpath_for_built_shared_libraries" ]
 }
 
-if (!build_with_chromium && is_clang) {
-  v8_remove_configs += [ "//build/config/clang:find_bad_constructs" ]
-}
+#if (!build_with_chromium && is_clang) {
+#  v8_remove_configs += [ "//build/config/clang:find_bad_constructs" ]
+#}
 
 # All templates should be kept in sync.
 template("v8_source_set") {
