diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index c411ee37c..218b2477c 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1433,7 +1433,7 @@ config("default_warnings") {
       # TODO(thakis): Remove this once
       # https://swiftshader-review.googlesource.com/c/SwiftShader/+/57968 has
       # rolled into angle.
-      cflags += [ "/wd4244" ]
+      # cflags += [ "/wd4244" ]
     }
   } else {
     if (is_apple && !is_nacl) {
@@ -1696,7 +1696,7 @@ config("no_chromium_code") {
     }
     cflags += [
       "/wd4800",  # Disable warning when forcing value to bool.
-      "/wd4267",  # TODO(jschuh): size_t to int.
+      #"/wd4267",  # TODO(jschuh): size_t to int.
     ]
   } else {
     # GCC may emit unsuppressible warnings so don't add -Werror for no chromium
diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
index 3a6dc1e54..74942f30b 100644
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -454,16 +454,16 @@ config("default_crt") {
     # Component mode: dynamic CRT. Since the library is shared, it requires
     # exceptions or will give errors about things not matching, so keep
     # exceptions on.
-    configs = [ ":dynamic_crt" ]
+    configs = [ ":dynamic_crt", ":win_msvc_cfg" ]
   } else {
     if (current_os == "winuwp") {
       # https://blogs.msdn.microsoft.com/vcblog/2014/06/10/the-great-c-runtime-crt-refactoring/
       # contains a details explanation of what is happening with the Windows
       # CRT in Visual Studio releases related to Windows store applications.
-      configs = [ ":dynamic_crt" ]
+      configs = [ ":dynamic_crt", ":win_msvc_cfg" ]
     } else {
       # Desktop Windows: static CRT.
-      configs = [ ":static_crt" ]
+      configs = [ ":dynamic_crt", ":win_msvc_cfg" ]
     }
   }
 }
@@ -592,3 +592,15 @@ config("lean_and_mean") {
 config("nominmax") {
   defines = [ "NOMINMAX" ]
 }
+
+# Control Flow Guard (CFG)
+config("win_msvc_cfg") {
+  if (!is_clang && !is_debug) {
+    cflags = [ "/guard:cf", "/Qspectre", "/W3" ]
+    ldflags = [ "/guard:cf" ]
+
+    if (current_cpu != "arm64") {
+      ldflags += [ "/CETCOMPAT" ]
+    }
+  }
+}
diff --git a/toolchain/win/setup_toolchain.py b/toolchain/win/setup_toolchain.py
index 831e36f74..6d9d295d4 100644
--- a/toolchain/win/setup_toolchain.py
+++ b/toolchain/win/setup_toolchain.py
@@ -176,6 +176,8 @@ def _LoadToolchainEnv(cpu, toolchain_root, sdk_dir, target_store):
     # building with a new and untested SDK. This should stay in sync with the
     # packaged toolchain in build/vs_toolchain.py.
     args.append('10.0.20348.0')
+    # Use Spectre runtime libraries
+    args.append('-vcvars_spectre_libs=spectre')
     variables = _LoadEnvFromBat(args)
   return _ExtractImportantEnvironment(variables)
 
