diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
index 51437c66c..728555267 100644
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -486,7 +486,7 @@ config("default_crt") {
       configs = [ ":dynamic_crt" ]
     } else {
       # Desktop Windows: static CRT.
-      configs = [ ":static_crt" ]
+      configs = [ ":dynamic_crt" ]
     }
   }
 }
@@ -594,3 +594,11 @@ config("lean_and_mean") {
 config("nominmax") {
   defines = [ "NOMINMAX" ]
 }
+
+# Control Flow Guard (CFG)
+config("win_msvc_cfg") {
+  if (!is_clang && !is_debug) {
+    cflags = [ "/guard:cf", "/Qspectre" ]
+    ldflags = [ "/guard:cf" ]
+  }
+}
\ No newline at end of file
