diff --git a/BUILD.gn b/BUILD.gn
index f0976f97e5..c4f6a026cf 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1511,7 +1511,7 @@ config("toolchain") {
   if (is_win) {
     cflags += [
       "/wd4245",  # Conversion with signed/unsigned mismatch.
-      "/wd4267",  # Conversion with possible loss of data.
+      #"/wd4267",  # Conversion with possible loss of data.
       "/wd4324",  # Padding structure due to alignment.
       "/wd4701",  # Potentially uninitialized local variable.
       "/wd4702",  # Unreachable code.
@@ -1615,14 +1615,14 @@ config("toolchain") {
 
       "/wd4100",  # Unreferenced formal function parameter.
       "/wd4121",  # Alignment of a member was sensitive to packing.
-      "/wd4244",  # Conversion: possible loss of data.
+      #"/wd4244",  # Conversion: possible loss of data.
       "/wd4505",  # Unreferenced local function has been removed.
       "/wd4510",  # Default constructor could not be generated.
       "/wd4512",  # Assignment operator could not be generated.
       "/wd4610",  # Class can never be instantiated, constructor required.
       "/wd4838",  # Narrowing conversion. Doesn't seem to be very useful.
-      "/wd4995",  # 'X': name was marked as #pragma deprecated
-      "/wd4996",  # Deprecated function warning.
+      #"/wd4995",  # 'X': name was marked as #pragma deprecated
+      #"/wd4996",  # Deprecated function warning.
 
       # These are variable shadowing warnings that are new in VS2015. We
       # should work through these at some point -- they may be removed from
@@ -1646,7 +1646,7 @@ config("toolchain") {
       "/wd4245",  # 'conversion' : conversion from 'type1' to 'type2',
                   # signed/unsigned mismatch
 
-      "/wd4267",  # 'var' : conversion from 'size_t' to 'type', possible loss of
+      #"/wd4267",  # 'var' : conversion from 'size_t' to 'type', possible loss of
                   # data
 
       "/wd4305",  # 'identifier' : truncation from 'type1' to 'type2'
@@ -7147,26 +7147,6 @@ group("v8_python_base") {
   data = [ ".vpython3" ]
 }
 
-group("v8_clusterfuzz") {
-  testonly = true
-
-  deps = [
-    ":d8",
-    ":v8_simple_inspector_fuzzer",
-    "tools/clusterfuzz/trials:v8_clusterfuzz_resources",
-  ]
-
-  if (v8_multi_arch_build) {
-    deps += [
-      ":d8(//build/toolchain/linux:clang_x64)",
-      ":d8(//build/toolchain/linux:clang_x64_v8_arm64)",
-      ":d8(//build/toolchain/linux:clang_x86)",
-      ":d8(//build/toolchain/linux:clang_x86_v8_arm)",
-      ":d8(tools/clusterfuzz/foozzie/toolchain:clang_x64_pointer_compression)",
-    ]
-  }
-}
-
 # Targets we ensure work with gcc. The aim is to keep this list small to have
 # a fast overall compile time.
 group("v8_gcc_light") {
@@ -7401,7 +7381,7 @@ v8_executable("d8") {
   }
 
   if (v8_correctness_fuzzer) {
-    deps += [ "tools/clusterfuzz/foozzie:v8_correctness_fuzzer_resources" ]
+    # deps += [ "tools/clusterfuzz/foozzie:v8_correctness_fuzzer_resources" ]
   }
 
   defines = []
@@ -8096,3 +8076,9 @@ if (!build_with_chromium && v8_use_perfetto) {
     ]
   }
 }  # if (!build_with_chromium && v8_use_perfetto)
+
+group("jsi") {
+  deps = [
+    "jsi:v8jsi",
+  ]
+}
diff --git a/DEPS b/DEPS
index 529f54b59f..f0ecac5c0c 100644
--- a/DEPS
+++ b/DEPS
@@ -708,4 +708,15 @@ hooks = [
                '--quiet',
                ],
   },
+  {
+    'name': 'rc_win',
+    'pattern': '.',
+    'condition': 'checkout_win and host_os == "win"',
+    'action': [ 'download_from_google_storage',
+                '--no_resume',
+                '--no_auth',
+                '--bucket', 'chromium-browser-clang/rc',
+                '-s', 'build/toolchain/win/rc/win/rc.exe.sha1',
+    ],
+  },
 ]
diff --git a/src/heap/cppgc/marking-state.h b/src/heap/cppgc/marking-state.h
index 4ce1ce4074..3153407f4f 100644
--- a/src/heap/cppgc/marking-state.h
+++ b/src/heap/cppgc/marking-state.h
@@ -342,7 +342,7 @@ class MutatorMarkingState final : public BasicMarkingState {
   ~MutatorMarkingState() override = default;
 
   inline bool MarkNoPush(HeapObjectHeader& header) {
-    return MutatorMarkingState::BasicMarkingState::MarkNoPush(header);
+    return BasicMarkingState::MarkNoPush(header);
   }
 
   inline void ReTraceMarkedWeakContainer(cppgc::Visitor&, HeapObjectHeader&);
diff --git a/src/maglev/maglev-ir.h b/src/maglev/maglev-ir.h
index 68a751b775..75277e0f7e 100644
--- a/src/maglev/maglev-ir.h
+++ b/src/maglev/maglev-ir.h
@@ -2335,8 +2335,8 @@ class NodeTMixin : public Base {
   template <typename... Args>
   explicit NodeTMixin(uint64_t bitfield, Args&&... args)
       : Base(bitfield, std::forward<Args>(args)...) {
-    DCHECK_EQ(this->NodeBase::opcode(), NodeBase::opcode_of<Derived>);
-    DCHECK_EQ(this->NodeBase::properties(), Derived::kProperties);
+    //DCHECK_EQ(this->NodeBase::opcode(), NodeBase::opcode_of<Derived>);
+    //DCHECK_EQ(this->NodeBase::properties(), Derived::kProperties);
   }
 };
 
@@ -2403,7 +2403,7 @@ class FixedInputNodeTMixin : public NodeTMixin<Base, Derived> {
   template <typename... Args>
   explicit FixedInputNodeTMixin(uint64_t bitfield, Args&&... args)
       : NodeTMixin<Base, Derived>(bitfield, std::forward<Args>(args)...) {
-    DCHECK_EQ(this->NodeBase::input_count(), kInputCount);
+    //DCHECK_EQ(this->NodeBase::input_count(), kInputCount);
   }
 };
 
diff --git a/src/utils/allocation.cc b/src/utils/allocation.cc
index dc8a3696fa..50bcba0a3b 100644
--- a/src/utils/allocation.cc
+++ b/src/utils/allocation.cc
@@ -64,8 +64,10 @@ const int kAllocationTries = 2;
 }  // namespace
 
 v8::PageAllocator* GetPlatformPageAllocator() {
-  DCHECK_NOT_NULL(GetPageAllocatorInitializer()->page_allocator());
-  return GetPageAllocatorInitializer()->page_allocator();
+  //DCHECK_NOT_NULL(GetPageAllocatorInitializer()->page_allocator());
+  //return GetPageAllocatorInitializer()->page_allocator();
+
+  return V8::GetCurrentPlatform()->GetPageAllocator();
 }
 
 v8::VirtualAddressSpace* GetPlatformVirtualAddressSpace() {
diff --git a/third_party/googletest/BUILD.gn b/third_party/googletest/BUILD.gn
index 1cf84b3d8f..d5c5ec9521 100644
--- a/third_party/googletest/BUILD.gn
+++ b/third_party/googletest/BUILD.gn
@@ -88,8 +88,8 @@ source_set("gtest") {
 
   all_dependent_configs = [ ":gtest_config" ]
 
-  configs -= [ "//build/config/compiler:chromium_code" ]
-  configs += [ "//build/config/compiler:no_chromium_code" ]
+  configs -= [ "//build/config/compiler:chromium_code", "//build/config/compiler:no_rtti", "//build/config/compiler:no_exceptions" ]
+  configs += [ "//build/config/compiler:no_chromium_code", "//build/config/compiler:rtti", "//build/config/compiler:exceptions" ]
 
   # V8-only workaround for http://crbug.com/chromium/1191946. Ensures that
   # googletest is compiled with the same visibility such as the rest of V8, see
diff --git a/tools/v8windbg/BUILD.gn b/tools/v8windbg/BUILD.gn
index 3bb1ee25e6..b1c31172d4 100644
--- a/tools/v8windbg/BUILD.gn
+++ b/tools/v8windbg/BUILD.gn
@@ -23,7 +23,7 @@ source_set("v8windbg_base") {
     "DbgEng.lib",
     "DbgModel.lib",
     "RuntimeObject.lib",
-    "comsuppwd.lib",
+    "comsuppw.lib",
   ]
 
   public_configs = [ ":v8windbg_config" ]
