diff --git a/BUILD.gn b/BUILD.gn
index dd97c4f922..785804f82d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1411,7 +1411,7 @@ config("toolchain") {
   } else if (target_os == "mac") {
     defines += [ "V8_HAVE_TARGET_OS" ]
     defines += [ "V8_TARGET_OS_MACOS" ]
-  } else if (target_os == "win") {
+  } else if (target_os == "win" || target_os == "winuwp") {
     defines += [ "V8_HAVE_TARGET_OS" ]
     defines += [ "V8_TARGET_OS_WIN" ]
   } else if (target_os == "chromeos") {
@@ -1490,7 +1490,7 @@ config("toolchain") {
   if (is_win) {
     cflags += [
       "/wd4245",  # Conversion with signed/unsigned mismatch.
-      "/wd4267",  # Conversion with possible loss of data.
+      #"/wd4267",  # Conversion with possible loss of data.
       "/wd4324",  # Padding structure due to alignment.
       "/wd4701",  # Potentially uninitialized local variable.
       "/wd4702",  # Unreachable code.
@@ -1594,14 +1594,14 @@ config("toolchain") {
 
       "/wd4100",  # Unreferenced formal function parameter.
       "/wd4121",  # Alignment of a member was sensitive to packing.
-      "/wd4244",  # Conversion: possible loss of data.
+      #"/wd4244",  # Conversion: possible loss of data.
       "/wd4505",  # Unreferenced local function has been removed.
       "/wd4510",  # Default constructor could not be generated.
       "/wd4512",  # Assignment operator could not be generated.
       "/wd4610",  # Class can never be instantiated, constructor required.
       "/wd4838",  # Narrowing conversion. Doesn't seem to be very useful.
-      "/wd4995",  # 'X': name was marked as #pragma deprecated
-      "/wd4996",  # Deprecated function warning.
+      #"/wd4995",  # 'X': name was marked as #pragma deprecated
+      #"/wd4996",  # Deprecated function warning.
 
       # These are variable shadowing warnings that are new in VS2015. We
       # should work through these at some point -- they may be removed from
@@ -1625,7 +1625,7 @@ config("toolchain") {
       "/wd4245",  # 'conversion' : conversion from 'type1' to 'type2',
                   # signed/unsigned mismatch
 
-      "/wd4267",  # 'var' : conversion from 'size_t' to 'type', possible loss of
+      #"/wd4267",  # 'var' : conversion from 'size_t' to 'type', possible loss of
                   # data
 
       "/wd4305",  # 'identifier' : truncation from 'type1' to 'type2'
@@ -6185,7 +6185,6 @@ v8_component("v8_libbase") {
     defines += [ "_CRT_RAND_S" ]  # for rand_s()
 
     libs = [
-      "dbghelp.lib",
       "winmm.lib",
       "ws2_32.lib",
     ]
@@ -6932,26 +6931,6 @@ group("v8_python_base") {
   data = [ ".vpython3" ]
 }
 
-group("v8_clusterfuzz") {
-  testonly = true
-
-  deps = [
-    ":d8",
-    ":v8_simple_inspector_fuzzer",
-    "tools/clusterfuzz/trials:v8_clusterfuzz_resources",
-  ]
-
-  if (v8_multi_arch_build) {
-    deps += [
-      ":d8(//build/toolchain/linux:clang_x64)",
-      ":d8(//build/toolchain/linux:clang_x64_v8_arm64)",
-      ":d8(//build/toolchain/linux:clang_x86)",
-      ":d8(//build/toolchain/linux:clang_x86_v8_arm)",
-      ":d8(tools/clusterfuzz/foozzie/toolchain:clang_x64_pointer_compression)",
-    ]
-  }
-}
-
 # Targets we ensure work with gcc. The aim is to keep this list small to have
 # a fast overall compile time.
 group("v8_gcc_light") {
@@ -7186,7 +7165,7 @@ v8_executable("d8") {
   }
 
   if (v8_correctness_fuzzer) {
-    deps += [ "tools/clusterfuzz/foozzie:v8_correctness_fuzzer_resources" ]
+    # deps += [ "tools/clusterfuzz/foozzie:v8_correctness_fuzzer_resources" ]
   }
 
   defines = []
@@ -7881,3 +7860,9 @@ if (!build_with_chromium && v8_use_perfetto) {
     ]
   }
 }  # if (!build_with_chromium && v8_use_perfetto)
+
+group("jsi") {
+  deps = [
+    "jsi:v8jsi",
+  ]
+}
diff --git a/DEPS b/DEPS
index 5cffcad442..f8264e6c44 100644
--- a/DEPS
+++ b/DEPS
@@ -680,4 +680,15 @@ hooks = [
     'condition': 'host_os == "win"',
     'action': ['python3', 'build/del_ninja_deps_cache.py'],
   },
+  {
+    'name': 'rc_win',
+    'pattern': '.',
+    'condition': 'checkout_win and host_os == "win"',
+    'action': [ 'download_from_google_storage',
+                '--no_resume',
+                '--no_auth',
+                '--bucket', 'chromium-browser-clang/rc',
+                '-s', 'build/toolchain/win/rc/win/rc.exe.sha1',
+    ],
+  }
 ]
diff --git a/gni/snapshot_toolchain.gni b/gni/snapshot_toolchain.gni
index c7faa8ca58..e041f7455f 100644
--- a/gni/snapshot_toolchain.gni
+++ b/gni/snapshot_toolchain.gni
@@ -74,6 +74,9 @@ if (v8_snapshot_toolchain == "") {
     # Cross-build from arm64 to intel (likely on an Apple Silicon mac).
     v8_snapshot_toolchain =
         "//build/toolchain/${host_os}:clang_arm64_v8_$v8_current_cpu"
+  } else if (target_os == "winuwp") {
+    # cross compile UWP Windows with host toolchain (for x64 and x86)
+    v8_snapshot_toolchain = host_toolchain
   } else if (host_cpu == "x64") {
     # This is a cross-compile from an x64 host to either a non-Intel target
     # cpu or to 32-bit x86 on a different target OS.
diff --git a/src/base/debug/stack_trace_win.cc b/src/base/debug/stack_trace_win.cc
index f981bec610..753438fa4a 100644
--- a/src/base/debug/stack_trace_win.cc
+++ b/src/base/debug/stack_trace_win.cc
@@ -29,6 +29,35 @@ namespace v8 {
 namespace base {
 namespace debug {
 
+#if 1 // dbghelp dependency unwanted
+
+bool EnableInProcessStackDumping() {
+  return false;
+}
+
+void DisableSignalStackDump() {
+}
+
+StackTrace::StackTrace() {
+}
+
+StackTrace::StackTrace(EXCEPTION_POINTERS* exception_pointers) {
+  InitTrace(exception_pointers->ContextRecord);
+}
+
+StackTrace::StackTrace(const CONTEXT* context) { InitTrace(context); }
+
+void StackTrace::InitTrace(const CONTEXT*) {
+}
+
+void StackTrace::Print() const { OutputToStream(&std::cerr); }
+
+void StackTrace::OutputToStream(std::ostream* os) const {
+    (*os) << "==== C stack trace not implemented for WINUWP ===============================\n";
+}
+
+#else // dbghelp dependency unwanted
+
 namespace {
 
 // Previous unhandled filter. Will be called if not nullptr when we intercept an
@@ -244,6 +273,8 @@ void StackTrace::OutputToStream(std::ostream* os) const {
   }
 }
 
+#endif // dbghelp
+
 }  // namespace debug
 }  // namespace base
 }  // namespace v8
diff --git a/src/base/platform/platform-win32.cc b/src/base/platform/platform-win32.cc
index 152fbbf3c2..51ba9b9cae 100644
--- a/src/base/platform/platform-win32.cc
+++ b/src/base/platform/platform-win32.cc
@@ -1422,8 +1422,8 @@ bool AddressSpaceReservation::DecommitPages(void* address, size_t size) {
 #define VOID void
 #endif
 
-// DbgHelp isn't supported on MinGW yet
-#ifndef __MINGW32__
+// DbgHelp isn't supported on Windows
+#if 0
 // DbgHelp.h functions.
 using DLL_FUNC_TYPE(SymInitialize) = BOOL(__stdcall*)(IN HANDLE hProcess,
                                                       IN PSTR UserSearchPath,
diff --git a/src/diagnostics/unwinding-info-win64.cc b/src/diagnostics/unwinding-info-win64.cc
index a71b866135..3731d04aa0 100644
--- a/src/diagnostics/unwinding-info-win64.cc
+++ b/src/diagnostics/unwinding-info-win64.cc
@@ -535,7 +535,7 @@ void RegisterNonABICompliantCodeRange(void* start, size_t size_in_bytes) {
   // by the embedder (like Crashpad).
 
   if (RegisterUnwindInfoForExceptionHandlingOnly()) {
-#if defined(V8_OS_WIN_X64)
+#if 0
     // Windows ARM64 starts since 1709 Windows build, no need to have exception
     // handling only unwind info for compatibility.
     if (unhandled_exception_callback_g) {
@@ -574,7 +574,7 @@ void UnregisterNonABICompliantCodeRange(void* start) {
   DCHECK(CanRegisterUnwindInfoForNonABICompliantCodeRange());
 
   if (RegisterUnwindInfoForExceptionHandlingOnly()) {
-#if defined(V8_OS_WIN_X64)
+#if 0
     // Windows ARM64 starts since 1709 Windows build, no need to have exception
     // handling only unwind info for compatibility.
     if (unhandled_exception_callback_g) {
diff --git a/src/maglev/maglev-ir.h b/src/maglev/maglev-ir.h
index 888944b606..7bf2c4c7e5 100644
--- a/src/maglev/maglev-ir.h
+++ b/src/maglev/maglev-ir.h
@@ -2258,8 +2258,8 @@ class NodeTMixin : public Base {
   template <typename... Args>
   explicit NodeTMixin(uint64_t bitfield, Args&&... args)
       : Base(bitfield, std::forward<Args>(args)...) {
-    DCHECK_EQ(NodeBase::opcode(), NodeBase::opcode_of<Derived>);
-    DCHECK_EQ(NodeBase::properties(), Derived::kProperties);
+    //DCHECK_EQ(NodeBase::opcode(), NodeBase::opcode_of<Derived>);
+    //DCHECK_EQ(NodeBase::properties(), Derived::kProperties);
   }
 };
 
@@ -2326,7 +2326,7 @@ class FixedInputNodeTMixin : public NodeTMixin<Base, Derived> {
   template <typename... Args>
   explicit FixedInputNodeTMixin(uint64_t bitfield, Args&&... args)
       : NodeTMixin<Base, Derived>(bitfield, std::forward<Args>(args)...) {
-    DCHECK_EQ(NodeBase::input_count(), kInputCount);
+    //DCHECK_EQ(NodeBase::input_count(), kInputCount);
   }
 };
 
diff --git a/src/objects/scope-info.cc b/src/objects/scope-info.cc
index 586c7a5d16..5083fcce8c 100644
--- a/src/objects/scope-info.cc
+++ b/src/objects/scope-info.cc
@@ -1043,6 +1043,8 @@ int ScopeInfo::ParametersStartIndex() const {
 }
 
 int ScopeInfo::FunctionContextSlotIndex(Tagged<String> name) const {
+  // This debug check in failing when running CPU profiler on some V8-JSI instances
+  // TODO: Investigate and find the root cause
   DCHECK(IsInternalizedString(name));
   if (HasContextAllocatedFunctionName()) {
     DCHECK_IMPLIES(HasFunctionName(), IsInternalizedString(FunctionName()));
diff --git a/src/snapshot/embedded/platform-embedded-file-writer-base.cc b/src/snapshot/embedded/platform-embedded-file-writer-base.cc
index e0602edc7e..4123b90e1b 100644
--- a/src/snapshot/embedded/platform-embedded-file-writer-base.cc
+++ b/src/snapshot/embedded/platform-embedded-file-writer-base.cc
@@ -138,7 +138,7 @@ EmbeddedTargetOs ToEmbeddedTargetOs(const char* s) {
     return EmbeddedTargetOs::kFuchsia;
   } else if (string == "ios" || string == "mac") {
     return EmbeddedTargetOs::kMac;
-  } else if (string == "win") {
+  } else if (string == "win" || string == "winuwp") {
     return EmbeddedTargetOs::kWin;
   } else if (string == "starboard") {
     return EmbeddedTargetOs::kStarboard;
diff --git a/src/trap-handler/trap-handler-simulator.h b/src/trap-handler/trap-handler-simulator.h
index 57471de1cb..1bd3bd0c3b 100644
--- a/src/trap-handler/trap-handler-simulator.h
+++ b/src/trap-handler/trap-handler-simulator.h
@@ -31,8 +31,10 @@ uintptr_t ProbeMemory(uintptr_t address, uintptr_t pc)
 // "ProbeMemory", but we want something more expressive on stack traces.
 #if V8_OS_DARWIN
     asm("_v8_internal_simulator_ProbeMemory");
-#else
+#elif !defined(_MSC_VER)
     asm("v8_internal_simulator_ProbeMemory");
+#else // MSVC
+    ;
 #endif
 
 }  // namespace v8::internal::trap_handler
diff --git a/src/utils/allocation.cc b/src/utils/allocation.cc
index dc8a3696fa..50bcba0a3b 100644
--- a/src/utils/allocation.cc
+++ b/src/utils/allocation.cc
@@ -64,8 +64,10 @@ const int kAllocationTries = 2;
 }  // namespace
 
 v8::PageAllocator* GetPlatformPageAllocator() {
-  DCHECK_NOT_NULL(GetPageAllocatorInitializer()->page_allocator());
-  return GetPageAllocatorInitializer()->page_allocator();
+  //DCHECK_NOT_NULL(GetPageAllocatorInitializer()->page_allocator());
+  //return GetPageAllocatorInitializer()->page_allocator();
+
+  return V8::GetCurrentPlatform()->GetPageAllocator();
 }
 
 v8::VirtualAddressSpace* GetPlatformVirtualAddressSpace() {
diff --git a/third_party/googletest/BUILD.gn b/third_party/googletest/BUILD.gn
index bc82c635da..648972c77a 100644
--- a/third_party/googletest/BUILD.gn
+++ b/third_party/googletest/BUILD.gn
@@ -88,8 +88,8 @@ source_set("gtest") {
 
   all_dependent_configs = [ ":gtest_config" ]
 
-  configs -= [ "//build/config/compiler:chromium_code" ]
-  configs += [ "//build/config/compiler:no_chromium_code" ]
+  configs -= [ "//build/config/compiler:chromium_code", "//build/config/compiler:no_rtti", "//build/config/compiler:no_exceptions" ]
+  configs += [ "//build/config/compiler:no_chromium_code", "//build/config/compiler:rtti", "//build/config/compiler:exceptions" ]
 
   # V8-only workaround for http://crbug.com/chromium/1191946. Ensures that
   # googletest is compiled with the same visibility such as the rest of V8, see
diff --git a/tools/v8windbg/BUILD.gn b/tools/v8windbg/BUILD.gn
index 5516a6109f..d6cde0082c 100644
--- a/tools/v8windbg/BUILD.gn
+++ b/tools/v8windbg/BUILD.gn
@@ -26,7 +26,7 @@ source_set("v8windbg_base") {
     "DbgEng.lib",
     "DbgModel.lib",
     "RuntimeObject.lib",
-    "comsuppwd.lib",
+    "comsuppw.lib",
   ]
 
   public_configs = [ ":v8windbg_config" ]
